File: sdk/javascript/src/resources/analytics.js
export class AnalyticsResource {
  constructor(http) {
    this.http = http;
  }

  /**
   * GET /analytics/{restaurant_id} — unified cross-platform metrics.
   * `from` and `to` are required (YYYY-MM-DD); `platform` filters to one platform.
   */
  get(restaurantId, { from, to, platform } = {}) {
    return this.http.get(`/analytics/${encodeURIComponent(restaurantId)}`, {
      query: { from, to, platform },
    });
  }
}

File: sdk/python/src/ufds_sdk/resources/analytics.py
from __future__ import annotations

from typing import Any, Optional
from urllib.parse import quote

from ..client import UFDSHttpClient


class AnalyticsResource:
    def __init__(self, http: UFDSHttpClient) -> None:
        self.http = http

    def get(
        self,
        restaurant_id: str,
        *,
        from_: Optional[str] = None,
        to: Optional[str] = None,
        platform: Optional[str] = None,
    ) -> Any:
        """GET /analytics/{restaurant_id} — unified cross-platform metrics.

        `from_` and `to` are required by the API (YYYY-MM-DD); `platform` filters to one platform.
        """
        return self.http.get(
            f"/analytics/{quote(restaurant_id, safe='')}",
            query={"from": from_, "to": to, "platform": platform},
        )
